### PR TITLE
add random seed to tf embedding

### PR DESCRIPTION
**Pull request by RolandJAAI**
Training the Tensorflow Embedding currently gives different results after each run, even if nothing else changed. This can lead to very unstable behavior especially with regards to unseen samples, but also for the training data. For example the user input "ich brauche zahnersatz" yields totally different results after two runs with exactly the same parameters and data. This sample is a part of a slightly longer sample which is in the training data:

rasa_core.processor  - Received user message 'ich brauche zahnersatz' with intent '{'name': 'zahnersatz_zuschuss', 'confidence': 0.909192681312561}' 

rasa_core.processor  - Received user message 'ich brauche zahnersatz' with intent '{'name': 'zahnersatz_zuschuss', 'confidence': 0.6224032044410706}'

The same goes for actual training data like "hi": 
rasa_core.processor  - Received user message 'hi' with intent '{'name': 'greet', 'confidence': 0.9807385206222534}'

rasa_core.processor  - Received user message 'hi' with intent '{'name': 'greet', 'confidence': 0.878046989440918}'

Setting reasonable fallback thresholds (like here at 0.9)  is therefore difficult. Each training run of a production system is a risk. 

This PR changes this and sets a stable random seed for Tensorflow and Numpy which can be configured in the pipeline settings.

**Proposed changes**:
- added config parameter random_seed to classifier
- added random seed initialization to train method of classifier
- updated changelog
- updated docs
- created test for random seed

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog

### CODE REVIEW

**Comment by akelad** [comment]
Thanks for submitting this PR, we'll give it a review as soon as possible

File: rasa_nlu/classifiers/embedding_intent_classifier.py [hunk 1]
```diff
@@ -87,7 +87,11 @@ class EmbeddingIntentClassifier(Component):
 
         # visualization of accuracy
         "evaluate_every_num_epochs": 10,  # small values may hurt performance
-        "evaluate_on_num_examples": 1000  # large values may hurt performance
+        "evaluate_on_num_examples": 1000,  # large values may hurt performance
+
+        # set random seed to get reproducible results
+        # try to change to another int if you are not getting good results
+        "random_seed": 1
```
Comment (Ghostvv) [inline rasa_nlu/classifiers/embedding_intent_classifier.py]: I think it is reasonable to set it to `None` to turn it off by default

**Comment by RolandJAAI** [comment]
Had to do some internal merging and re-branching, will open the version with requested changes as new PR
